Remotion/ObjectBinding/Web.ClientScript/Scripts/BocBooleanValue.ts

```diff
@@ -0,0 +1,187 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+
+//  BocBooleanValue.js contains client side scripts used by BocBooleanValue.
+
+var _bocBooleanValue_Resources: Dictionary<BocBooleanValue_Resource> = {};
+
+class BocBooleanValue_Resource
+{
+  constructor(
+    private readonly _trueValue: string,
+    private readonly _falseValue: string,
+    private readonly _nullValue: string,
+    private readonly _trueDescription: string,
+    private readonly _falseDescription: string,
+    private readonly _nullDescription: string,
+    private readonly _trueIconUrl: string,
+    private readonly _falseIconUrl: string,
+    private readonly _nullIconUrl: string) 
+  {
+  }
+
+  public SelectNextCheckboxValue (
+    link: HTMLElement,
+    icon: HTMLImageElement,
+    label: Nullable<HTMLElement>,
+    hiddenField: HTMLInputElement,
+    isRequired: boolean,
+    trueDescription: Nullable<string>,
+    falseDescription: Nullable<string>,
+    nullDescription: Nullable<string>) 
+  {
+    var trueValue = this._trueValue;
+    var falseValue = this._falseValue;
+    var nullValue = this._nullValue;
+
+    var oldValue = hiddenField.value;
+    var newValue;
+
+    //  Select the next value.
+    //  true -> false -> null -> true
+    if (isRequired) {
+        if (oldValue == falseValue)
+            newValue = trueValue;
+        else if (oldValue == nullValue)
+            newValue = trueValue;
+        else
+            newValue = falseValue;
+    }
+    else {
+        if (oldValue == falseValue)
+            newValue = nullValue;
+        else if (oldValue == nullValue)
+            newValue = trueValue;
+        else
+            newValue = falseValue;
+    }
+
+    // Update the controls
+    hiddenField.value = newValue;
+    var checkedState: Optional<string>;
+    var iconSrc: Optional<string>;
+    var description: Optional<string>;
+
+    if (newValue == falseValue) {
+        checkedState = 'false';
+        iconSrc = this._falseIconUrl;
+        if (falseDescription == null)
+            description = this._falseDescription;
+        else
+            description = falseDescription;
+    }
+    else if (newValue == nullValue) {
+        checkedState = 'mixed';
+        iconSrc = this._nullIconUrl;
+        if (nullDescription == null)
+            description = this._nullDescription;
+        else
+            description = nullDescription;
+    }
+    else if (newValue == trueValue) {
+        checkedState = 'true';
+        iconSrc = this._trueIconUrl;
+        if (trueDescription == null)
+            description = this._trueDescription;
+        else
+            description = trueDescription;
+    } // TODO: add default case to prevent variables from being undefined
```

Comment:
can do. Would be an exception case since we only support three values passed in the ctor.